Source/Mosa.DeviceSystem/FrameBuffer.cs

```diff
@@ -113,22 +113,59 @@ public void DrawImage(Image image, uint x, uint y, int transparentColor)
 		/// <param name="image">The image.</param>
 		/// <param name="x">X of the top left of the image.</param>
 		/// <param name="y">Y of the top left of the image.</param>
-		public void DrawImage(Image image, uint x, uint y)
+		public void DrawImage(Image image, uint x, uint y, bool alpha = false)
 		{
-			var wi = Math.Clamp(image.Width, 0, Width - x);
-			var he = Math.Clamp(image.Height, 0, Height - y);
+			if (alpha)
+			{
+				// Slow, find faster method (maybe?)
 
-			if (x < 0 || y < 0 || x >= Width || y >= Height)
-				return;
+				var wi = Math.Clamp(image.Width, 0, Width - x);
+				var he = Math.Clamp(image.Height, 0, Height - y);
 
-			for (int h = 0; h < he; h++)
-			{
-				for (int w = 0; w < wi; w++)
+				if (x < 0 || y < 0 || x >= Width || y >= Height)
+					return;
+
+				for (int h = 0; h < he; h++)
 				{
-					var color = image.GetColor(w, h);
+					for (int w = 0; w < wi; w++)
+					{
+						uint xx = (uint)(x + w);
+						uint yy = (uint)(y + h);
+
+						var color = image.GetColor(w, h);
+
+						Color foreground = Color.FromArgb(color);
+						Color background = Color.FromArgb((int)GetPixel(xx, yy));
+
+						int alphac = foreground.A;
+						int inv_alpha = 255 - alphac;
 
-					SetPixel((uint)color, (uint)(x + w), (uint)(y + h));
+						byte newR = (byte)(((foreground.R * alphac + inv_alpha * background.R) >> 8) & 0xFF);
+						byte newG = (byte)(((foreground.G * alphac + inv_alpha * background.G) >> 8) & 0xFF);
+						byte newB = (byte)(((foreground.B * alphac + inv_alpha * background.B) >> 8) & 0xFF);
+
+						SetPixel((uint)Color.ToArgb(newR, newG, newB), xx, yy);
+					}
+				}
+			}
+			else
+			{
+				int wb = image.Width * 4;
+				uint count = (uint)Math.Clamp(wb, 0, (Width - x) * 4);
+
+				Pointer imagePtr;
+
+				unsafe
```

Comment:
Avoid unsafe code.

Reply:
I don't know any other way to get a pointer from an array other than using fixed

Reply:
It's attempting to peek into a design detail. Use the public methods instead.

Reply:
Intrinsic.GetObjectAddress returns an incorrect value, it can't be used.

Reply:
Use the public methods of the Image class.

Reply:
But then DrawImage() is slow. It's actually the method that makes the biggest lost in performance (in Framebuffer at least)